Description: Disable chunking (BDAT) by default.
  Change default value of main option chunking_advertise_hosts and smtp
  transport option hosts_try_chunking from "*" to empty.
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Forwarded: not-needed
Last-Update: 2017-01-19

--- a/doc/spec.txt
+++ b/doc/spec.txt
@@ -13224,9 +13224,9 @@
 preceding 4.88 had these disabled by default; high-rate installations confident
 they will never run out of resources may wish to deliberately disable them.
 
-+--------------------------------------------------------------+
-|chunking_advertise_hosts|Use: main|Type: host list*|Default: *|
-+--------------------------------------------------------------+
++------------------------------------------------------------------+
+|chunking_advertise_hosts|Use: main|Type: host list*|Default: unset|
++------------------------------------------------------------------+
 
 The CHUNKING extension (RFC3030) will be advertised in the EHLO message to
 these hosts. Hosts may use the BDAT command as an alternate to DATA.
@@ -22540,9 +22540,9 @@
 unauthenticated. See also hosts_require_auth, and chapter 33 for details of
 authentication.
 
-+--------------------------------------------------------+
-|hosts_try_chunking|Use: smtp|Type: host list*|Default: *|
-+--------------------------------------------------------+
++------------------------------------------------------------+
+|hosts_try_chunking|Use: smtp|Type: host list*|Default: unset|
++------------------------------------------------------------+
 
 This option provides a list of servers to which, provided they announce
 CHUNKING support, Exim will attempt to use BDAT commands rather than DATA. BDAT
--- a/src/globals.c
+++ b/src/globals.c
@@ -500,7 +500,7 @@
 int     check_spool_inodes     = 100;
 int     check_spool_space      = 10*1024;	/* 10K Kbyte == 10MB */
 
-uschar *chunking_advertise_hosts = US"*";
+uschar *chunking_advertise_hosts = NULL;
 unsigned chunking_datasize     = 0;
 unsigned chunking_data_left    = 0;
 BOOL    chunking_offered       = FALSE;
--- a/src/transports/smtp.c
+++ b/src/transports/smtp.c
@@ -200,7 +200,7 @@
   NULL,                /* serialize_hosts */
   NULL,                /* hosts_try_auth */
   NULL,                /* hosts_require_auth */
-  US"*",               /* hosts_try_chunking */
+  NULL,                /* hosts_try_chunking */
 #ifdef EXPERIMENTAL_DANE
   NULL,                /* hosts_try_dane */
   NULL,                /* hosts_require_dane */
